src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1201,14 +1194,48 @@ class ChooseStringFromDataset : public Generator<std::string> {
             }
         }
     }
-
 private:
-    DefaultRandom& _rng;
-    ActorId _id;
     std::string _path;
     static inline DataSetCache _datasets;
-    bool _sequential;
-    uint64_t _i;
+    const std::vector<std::string>* _dataset;
+};
+
+class ChooseStringFromDatasetRandomly : public Generator<std::string>, private WithDatasetStorage {
+public:
+    ChooseStringFromDatasetRandomly(const Node& node, GeneratorArgs generatorArgs)
+        : WithDatasetStorage(node["path"].maybe<std::string>().value()),
+          _rng{generatorArgs.rng} {}
+
+    std::string evaluate() {
+        const auto& dataset = getDataset();
```

Comment:
I am very happy with how this and the next evaluate function turned out. Thank you.